Patches the email application to prompt for installation.

From: Rod Whitby <rod@whitby.id.au>

Tested-On: 1.1

It involves modifying/replacing the handleLinkClicked function.

http://predev.wikidot.com/email-app-patch-to-prompt-for-ipk-installation

Patch by simplyflipflops.
---

 .../app/controllers/message-assistant.js           |   40 +++++++++++++++++---
 1 files changed, 34 insertions(+), 6 deletions(-)


Index: /usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -1041,12 +1041,40 @@ var MessageAssistant = Class.create({
 	 * User clicked on a hyperlink.
 	 */
 	handleLinkClicked: function(event) {
-		Mojo.Log.info("handleLinkClicked %s", event.url);
-		this.controller.serviceRequest('palm://com.palm.applicationManager',
-				{
-					method: 'open',
-					parameters: {target: event.url}
-				});
+	  Mojo.Log.info("handleLinkClicked %s", event.url);
+
+	  var extensionIndex2 = event.url.lastIndexOf('.');
+	  if (extensionIndex2 > 0 ){
+	    var extensionTemp = event.url.substring(extensionIndex2 + 1).toLowerCase();
+	  }
+
+	  if (extensionTemp === "ipk") {
+	    this.controller.showAlertDialog({
+		onChoose: function(value) { 
+		  if (value===true) {
+		    this.controller.serviceRequest('palm://com.palm.applicationManager',
+			{
+			  method: 'open',
+			  parameters: {target: event.url}
+			});
+		  }
+		},
+
+		message: $L("Found Palm Application Bundle IPK. You should only install packages from trusted sources."),
+
+		choices: 
+		  [
+		    {label:$L('I Trust This Source - Install'), value: true, type:'affirmative'},
+		    {label:$L('Cancel Install'), value: false, type:'alert'}
+		  ]
+	    });
+	  } else {
+	    this.controller.serviceRequest('palm://com.palm.applicationManager',
+		{
+		  method: 'open',
+		  parameters: {target: event.url}
+		});
+	  }
 	},
 
 	/**
